+bids/copy_to_derivative.m

```diff
@@ -285,39 +282,58 @@ function copy_with_symlink(src, target, verbose)
   % Though datalad should run on windows too
   %
 
-  command = 'cp -R -L -f';
-
   if verbose
     fprintf(1, '\n copying %s --> %s', src, target);
   end
 
   if  isunix
-    status = system( ...
-                    sprintf('%s %s %s', ...
-                            command, ...
-                            src, ...
-                            target));
+
+    if unzip_files && is_gunzipped(src)
+      command = sprintf('gunzip -kfc %s > %s', src, target(1:end - 3));
+    else
+      command = sprintf('cp -R -L -f %s %s', src, target);
+    end
+
+    status = system(command);
+
     if status > 0
       msg = ['Copying data with system command failed: ' ...
              'Will use matlab/octave copyfile command instead.\n', ...
              'May be an issue if your data set contains symbolic links' ...
              '(e.g. if you use datalad or git-annex.)'];
       bids.internal.error_handling(mfilename, 'copyError', msg, true, verbose);
-      use_copyfile(src, target, verbose);
+
+      use_copyfile(src, target, unzip_files, verbose);
```

Comment:
I doubt that if the system copy fails, the matlab one will work

Reply:
good point

Reply:
implementing this now and testing: will merge with no further review once all test spass